src/utils/archive.ts

```diff
@@ -199,7 +200,59 @@ export async function createPublicDrive(
     txs: [driveTx, rootFolderTx]
   };
 }
+export async function sendArDriveFee(
+  keyfile: JWKInterface,
+  arPrice: number,
+  arweave: Arweave
+) {
+  try {
+    // Get the latest ArDrive Community Fee from the Community Smart Contract
+    let fee = arPrice * ((await getArDriveFee()) / 100);
+
+    // If the fee is too small, we assign a minimum
+    if (fee < 0.00001) {
+      fee = 0.00001;
+    }
+
+    // Probabilistically select the PST token holder
+    const holder = await selectTokenHolder();
+
+    // send a fee. You should inform the user about this fee and amount.
+    const transaction = await arweave.createTransaction(
+      {
+        target: holder,
+        quantity: arweave.ar.arToWinston(fee.toString())
+      },
+      keyfile
+    );
 
+    // Tag file with data upload Tipping metadata
+    transaction.addTag("App-Name", ardriveClient);
```

Comment:
Take a look at @vilenarios 's specs for these values in the ticket again and make sure they're what's requested. We may need to update some constant values at the top of the file.

Reply:
[Updated](https://github.com/th8ta/ArConnect/pull/33/commits/f76faa99ec9bfb48f340cbe9b4857e805e4e2582)